stag-library-compiler/src/main/java/com/vimeo/stag/processor/generators/model/accessor/FieldAccessor.java

```diff
@@ -0,0 +1,182 @@
+package com.vimeo.stag.processor.generators.model.accessor;
+
+import com.google.gson.annotations.JsonAdapter;
+import com.google.gson.annotations.SerializedName;
+import com.vimeo.stag.processor.utils.Preconditions;
+
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import javax.lang.model.element.AnnotationMirror;
+import javax.lang.model.element.Element;
+import javax.lang.model.element.VariableElement;
+import javax.lang.model.type.MirroredTypeException;
+import javax.lang.model.type.TypeMirror;
+
+/**
+ * A class representing a field that must be
+ * populated from JSON.
+ * <p>
+ * Created by restainoa on 5/8/17.
+ */
+public abstract class FieldAccessor {
+
+    @NotNull
+    private final VariableElement mVariableElement;
+
+    public FieldAccessor(@NotNull VariableElement element) {
+        mVariableElement = element;
+    }
+
+    /**
+     * Create the code that reads from the field.
+     * In the format:
+     * <p>
+     * <code>
+     * object.{getterCode};
+     * </code>
+     *
+     * @return a not null string that represents
+     * code that can be used to access the field.
+     */
+    @NotNull
+    public abstract String createGetterCode();
+
+    /**
+     * Create the code that assigns the field
+     * to the assigned code. In the format:
+     * <p>
+     * <code>
+     * object.{setterCode};
+     * </code>
+     *
+     * @param assignment the code that this field should
+     *                   be assigned to.
+     * @return a not null string that represents code
+     * that can be used to assign the field.
+     */
+    @NotNull
+    public abstract String createSetterCode(@NotNull String assignment);
+
+    /**
+     * The variable name of this field.
+     *
+     * @return the variable name of this field.
+     */
+    @NotNull
+    protected final String getVariableName() {
+        return mVariableElement.getSimpleName().toString();
+    }
+
+    /**
+     * Determines if this field was marked with
+     * a not null annotation that requires the
+     * field to be populated.
+     *
+     * @return true if the field was marked not null,
+     * false otherwise.
+     */
+    public final boolean requireNotNull() {
```

Comment:
This actually brings up another code style thing. Because it could alternatively be `doesRequireNotNull()` - which is more consistent with how we usually name booleans (and goes along with [this SO post](https://stackoverflow.com/questions/3874350/naming-conventions-for-java-methods-that-return-booleanno-question-mark)).

But [this](https://stackoverflow.com/questions/1566745/boolean-method-naming-readability) and [this](https://stackoverflow.com/questions/1370840/naming-conventions-what-to-name-a-method-that-returns-a-boolean) are also interesting answers.

We should add the result to code style regardless.